yt_dlp/extractor/aiyifan.py

```diff
@@ -0,0 +1,109 @@
+import hashlib
+import re
+import time
+
+from yt_dlp.extractor.common import InfoExtractor
+
+
+class AiyifanIE(InfoExtractor):
+    _VALID_URL = r'https?://(?:www\.)?yfsp\.tv/play/(?P<id>[^/?#]+)(?:\?id=(?P<alt_id>[^&#]+))?'
+    # All the three domains are the same
+    _TESTS = [
+        {
+            'url': 'https://www.yfsp.tv/play/tFAWlkx5kr9?id=GB7vRUxjOn5',
+            'info_dict': {
+                'id': 'GB7vRUxjOn5',
```

Comment:
Does passing `https://www.yfsp.tv/play/tFAWlkx5kr9` work without the `id` query parameter? If so, it's a problem. A given video should only have a single, unique ID. The ID shouldn't vary depending on what URL format is used